presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -163,6 +163,12 @@ private RelationPlan createExplainAnalyzePlan(Analysis analysis, Explain stateme
     {
         Scope scope = analysis.getScope(statement);
         PlanNode root = planStatement(analysis, statement.getStatement());
+        if (root instanceof OutputNode) {
+            // this extra output node was added by a call to plan underlying query and is not needed
```

Comment:
The sequence of actions is the following:

1. planStatement() is called for the initial query (which includes explain analyze).
2. createOutputPlan(createExplainAnalyzePlan(...)) is called.
3. createExplainAnalyzePlan() calls planStatement() again with the underlying query (select ...).
4. createOutputPlan(createRelationPlan(...)) is called which inserts output node for the underlying query. Here's where the second (incorrect) output node is added. It's incorrect, because it uses analyzer's output information to create outputs and it's not correct for this intermediate output node. It seems like there's an implicit assumption that there's only one OutputNode per plan.
5. createExplainAnalyzePlan() receives the underlying query plan (with the last node being OutputNode) and appends ExplainAnalyzeNode to it.
6. createOutputPlan() is called for the original explain analyze plan.

I'm going to upload a change with a different approach to fix it. See if it's better.